server/src/connectionHandlers/onDefinition.ts

```diff
@@ -72,14 +73,37 @@ export function onDefinitionHandler (textDocumentPositionParams: TextDocumentPos
     }
     // Symbols in string content
     if (analyzer.isStringContent(documentUri, position.line, position.character)) {
-      const allSymbolsAtPosition = analyzer.getSymbolsInStringContent(documentUri, position.line, position.character)
+      const wholeWordRegex = /(?<![-.:])\b(\w+)\b(?![-.:])/g
+      const uriRegex = /(file:\/\/)(?<uri>.*)\b/g
+      const [uriAtPosition] = analyzer.getSymbolsInStringContent(documentUri, position.line, position.character, uriRegex)
+      if (uriAtPosition !== undefined) {
+        const { workspaceFolders } = analyzer
+        if (workspaceFolders !== undefined && workspaceFolders !== null) {
+          for (const workspaceFolder of workspaceFolders) {
+            const filePath = findFileInDirectory(workspaceFolder.uri.replace('file://', ''), uriAtPosition.name)
```

Comment:
The user's workspace may contain several yocto layers and build directories which are not ignored here. Looking up the whole workspaceFolder will be inaccurate and slow. I suggest rather searching only the directory or immediate parent of the recipe file.